client/src/components/Overview.jsx

```diff
@@ -2,27 +2,59 @@ import axios from 'axios';
 import React from "react";
 import { connect } from 'react-redux';
 //import { FaFacebookSquare, FaInstagram, FaPinterest, FaTwitter } from 'react-icons/fa' ;
-//import { BsStar, BsStarHalf, BsStarFill } from 'react-icons/bs' ;
+import { BsStar, BsStarHalf, BsStarFill } from 'react-icons/bs';
 import StyleSelector from "./StyleSelector";
 import AddToCart from './AddtoCart';
 
 
 function setDefaultStyle(styles) {
-
+  //console.log(styles);
   const result = styles.filter((style) => style['default?'] === true);
- 
   return result;
 }
 
-class Overview extends React.Component {
- 
+//Finds and returns average of reviews
+//TODO: Handle edge case for no reviews in accordance with business docs
+function reviewAverage(ratings) {
+  let total = 0;
+  let weightedSum = 0;
+  for (let i = 1; i < 6; i++) {
+    // eslint-disable-next-line radix
+    total += parseInt(ratings[i]);
+    weightedSum += i * ratings[i];
+  }
+  return (weightedSum / total);
+}
 
+function generateStars(average) {
+  const filledStars = Math.trunc(average);
+  const filledRemainder = average % 1;
+  const emptyStars = 5 - filledStars;
+  const starBar = [];
+  for (let i = 0; i < filledStars; i++) {
+    starBar.push(<BsStarFill />);
+  } 
+  
+  if (filledRemainder >= 0.5) {
+    starBar.push(<BsStarHalf />);
+  }
+  
+  for (let i = 0; i < emptyStars; i++) {
+    starBar.push(<BsStar />);
+  } 
+
+  return starBar;
+}
+class Overview extends React.Component {
+  
 
   componentDidMount() {
-    const { setStyles, setSelectedStyle, setProductInfo } = this.props;
+    const {
+      setStyles, setSelectedStyle, setProductInfo, setMetaData
+    } = this.props;
```

Comment:
so for now, we are mounting and immediately assigning this product as the focus. Any  thoughts about how we are going to be reinitializing when the user pick a diff item and how we are going to be passing that data during re render?